homework/index.js

```diff
@@ -29,23 +31,127 @@
     return elem;
   }
 
+  function addRow(table, title, repoText) {
+    const tr = createAndAppend('tr', table);
+    createAndAppend('th', tr, {
+      text: title,
+    });
+    createAndAppend('td', tr, { text: repoText });
+    return tr;
+  }
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const date = new Date(repo.updated_at);
+    const li = createAndAppend('li', ul, { class: 'li-root' });
+    const table = createAndAppend('table', li);
+    const firstRow = addRow(table, 'Repository:', '');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: '_blank',
+    });
+
+    addRow(table, 'Description:', repo.description, 'td');
+    addRow(table, 'Forks:', repo.forks, 'td');
+    addRow(table, 'Updated:', date.toLocaleDateString(), 'td');
+  }
+
+  function createOptions(repo, parent, index) {
+    createAndAppend('option', parent, {
+      text: repo.name,
+      value: index,
+    });
+  }
+  function selections(selection, data, parent) {
+    const ul = createAndAppend('ul', parent, {
+      class: 'ul-root',
+    });
+    const repo = data[selection];
+    renderRepoDetails(repo, ul);
```

Comment:
Now that you are rendering the details for a single repository rather than a list of repositories as you did in week 1, it is now redundant to wrap your HTML table in an `ul`. This is what your code generates now:

```html
<section class="repo-container">
  <ul class="ul-root">
    <li class="li-root">
      <table>
        <!-- etc -->
      </table>
    </li>
  </ul>
</section>
```

But all you need is:

```html
<section class="repo-container">
  <table>
    <!-- etc -->
  </table>
</section>

```

Reply:
I will remove ul, li but add div inside this section because I need another child for styling this part in CSS.